dev/PowerNotifications/PowerNotifications.h

```diff
@@ -0,0 +1,709 @@
+            {
+                return PowerSchemePersonality::HighPerformance;
+            }
+            else
+            {
+                return PowerSchemePersonality::Balanced;
+            }
+
+        }
+
+        event_token PowerSchemePersonalityChanged(PowerEventHandle const& handler)
+        {
+            return AddCallback(powerSchemePersonalityFunc, handler);
+        }
+
+        void PowerSchemePersonalityChanged(event_token const& token)
+        {
+            RemoveCallback(powerSchemePersonalityFunc, token);
+        }
+
+        void PowerSchemePersonalityChanged_Callback(GUID powerSchemePersonality)
+        {
+            m_cachedPowerSchemePersonality = powerSchemePersonality;
+            RaiseEvent(powerSchemePersonalityFunc);
+        }
+
+
+        // UserPresenceStatus Functions
+        winrt::Microsoft::ProjectReunion::UserPresenceStatus UserPresenceStatus()
+        {
+            CheckRegistrationAndOrUpdateValue(userPresenceStatusFunc);
+            return static_cast<winrt::Microsoft::ProjectReunion::UserPresenceStatus>(m_cachedUserPresenceStatus);
+        }
+
+        event_token UserPresenceStatusChanged(PowerEventHandle const& handler)
+        {
+            return AddCallback(userPresenceStatusFunc, handler);
+        }
+
+        void UserPresenceStatusChanged(event_token const& token)
+        {
+            RemoveCallback(userPresenceStatusFunc, token);
+        }
+
+        void UserPresenceStatusChanged_Callback(DWORD userPresenceStatus)
+        {
+            m_cachedUserPresenceStatus = userPresenceStatus;
+            RaiseEvent(userPresenceStatusFunc);
+        }
+
+
+        // SystemAwayModeStatus Functions
+        winrt::Microsoft::ProjectReunion::SystemAwayModeStatus SystemAwayModeStatus()
+        {
+            CheckRegistrationAndOrUpdateValue(systemAwayModeStatusFunc);
+            return static_cast<winrt::Microsoft::ProjectReunion::SystemAwayModeStatus>(m_cachedSystemAwayModeStatus);
+        }
+
+        event_token SystemAwayModeStatusChanged(PowerEventHandle const& handler)
+        {
+            return AddCallback(systemAwayModeStatusFunc, handler);
+        }
+
+        void SystemAwayModeStatusChanged(event_token const& token)
+        {
+            RemoveCallback(systemAwayModeStatusFunc, token);
+        }
+
+        void SystemAwayModeStatusChanged_Callback(DWORD systemAwayModeStatus)
+        {
+            m_cachedSystemAwayModeStatus = systemAwayModeStatus;
+            RaiseEvent(systemAwayModeStatusFunc);
+        }
+
+
+    };
+
+};
+
+namespace winrt::Microsoft::ProjectReunion::implementation
+{
+
+     struct PowerManager
+     {
+        PowerManager() = default;
+
+        //Get function forwards
+        static winrt::Microsoft::ProjectReunion::EnergySaverStatus EnergySaverStatus()
+        {
+            return make_self<factory_implementation::PowerManager>()->EnergySaverStatus();
```

Comment:
This doesn't seem right - every method call here creates a new instance of PowerManager, calls one method, then discards it.

Reply:
It works because `factory_implementation::PowerManager` is marked `static_lifetime`, so each `make_self` just returns the same instance back.